service/src/main/java/bio/terra/landingzone/library/landingzones/management/deleterules/AKSAgentPoolHasMoreThanOneNode.java

```diff
@@ -0,0 +1,30 @@
+package bio.terra.landingzone.library.landingzones.management.deleterules;
+
+import static bio.terra.landingzone.library.landingzones.management.AzureResourceTypeUtils.AZURE_AKS_TYPE;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.management.ResourceToDelete;
+
+public class AKSAgentPoolHasMoreThanOneNode extends ResourceDependencyDeleteRule {
+  public AKSAgentPoolHasMoreThanOneNode(ArmManagers armManagers) {
+    super(armManagers);
+  }
+
+  @Override
+  public String getExpectedType() {
+    return AZURE_AKS_TYPE;
+  }
+
+  @Override
+  public boolean hasDependentResources(ResourceToDelete resourceToDelete) {
```

Comment:
Thinking out loud, wonder if this needs adjustments if/when we start managing user nodepools within a cluster.

Reply:
Yeah, agree, this rule could change as we evolve the definition of "usage" for AKS.  Something I considered was to check if the cluster was "stopped"; however, we aren't stopping the cluster yet in the deployment. Once we get to that, we can consider replacing this rule with a new one that checks the on/off state.